src/utils/gaa.js

```diff
@@ -2165,7 +2189,22 @@ export class GaaMetering {
       }
     }
 
-    return true;
+    if ('paywallReason' in newUserState) {
+      if (newUserState.granted) {
+        debugLog(
+          'userState.granted must be false when paywallReason is supplied.'
+        );
+        noIssues = false;
+      }
+
+      if (PaywallReasonType[newUserState.paywallReason] === undefined) {
+        debugLog(
+          'userState.paywallReason has to be empty or set to RESERVED_USER.'
+        );
+        noIssues = false;
+      }
+    }
+    return noIssues;
   }
```

Comment:
Not blocking for this PR but this file is getting quite large. We really should think about chunking it up

Reply:
Yeah, it would be good to slim it down and remove redundant code